starter-code/coffe-books/public/javascripts/map.js

```diff
@@ -0,0 +1,53 @@
+function main () {
+  const mapElement = document.getElementById('map');
+
+  function getBrowserLocation () {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+        const center = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        console.log('center: ', center);
+        startMap(center);
+        return center;
+      }, function () {
+        console.log('Error in the geolocation service.');
+      });
+    } else {
+      console.log('Browser does not support geolocation.');
+    }
+  }
+  function startMap (center) {
+    var map = new google.maps.Map(
+      mapElement,
+      {
+        zoom: 15,
+        center: center
+      }
+    );
+    // add marker
+    let markers = [];
+    const allPlaces = getData();
+    allPlaces.forEach((place) => {
```

Comment:
It is not displaying the pins

Reply:
As this code runs asynchronously it will continue with the following lines without having allPlaces defined by the function getData() (which is running asynchronously and not finished). You can put all the other stuff inside another function to avoid this asynchronous issue.